src/bin/stratis_dumpmetadata.rs

```diff
@@ -27,15 +27,18 @@ fn run(devpath: &str) -> Result<(), String> {
 
     let loaded_state = bda
         .load_state(&mut devfile)
-        .map_err(|stateload_err| format!("Error during load state: {}", stateload_err))?
-        .ok_or_else(|| "No BDA state found".to_string())?;
+        .map_err(|stateload_err| format!("Error during load state: {}", stateload_err))?;
     println!("State JSON data:");
```

Comment:
I think a better title than "State JSON data" would be "pool metadata", because that's really what it is. I've been calling it "variable length metadata", because that's how we have to deal with it, but "pool metadata" is a far more descriptive term for the user. Can you change this heading to "Pool metadata:", and where there is none just report "None" or "None found" or something like that?

Note that it will be hard to demonstrate the situation where the pool metadata isn't there w/out making a very large pool or crashing the pool with very precise timing. But it will occur eventually.

Note that you could actually use this script, with some alteration, to detect when the pool metadata gets written to a device. I've been somewhat worried about delays under certain conditions, and there's an issue out for that.